test/system/src/main/java/io/pravega/test/system/framework/services/docker/PravegaSegmentStoreDockerService.java

```
@@ -0,0 +1,143 @@
+import com.spotify.docker.client.messages.swarm.Resources;
+import com.spotify.docker.client.messages.swarm.ServiceMode;
+import com.spotify.docker.client.messages.swarm.ServiceSpec;
+import com.spotify.docker.client.messages.swarm.TaskSpec;
+import lombok.extern.slf4j.Slf4j;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import static org.hamcrest.core.Is.is;
+import static org.hamcrest.core.IsNull.notNullValue;
+import static org.junit.Assert.assertThat;
+
+@Slf4j
+public class PravegaSegmentStoreDockerService extends DockerBasedService {
+
+    private static final int SEGMENTSTORE_PORT = 12345;
+    private static final String SEGMENTSTORE_EXTRA_ENV = System.getProperty("segmentStoreExtraEnv");
+    private static final String ENV_SEPARATOR = ";;";
+    private static final java.lang.String KEY_VALUE_SEPARATOR = "::";
+    private int instances = 1;
+    private double cpu = 0.1;
+    private double mem = 1000.0;
+
+    public PravegaSegmentStoreDockerService(final String serviceName) {
+        super(serviceName);
+    }
+
+    @Override
+    public void stop() {
+        try {
+            dockerClient.removeService(getID());
+        } catch (DockerException | InterruptedException e) {
+            log.error("unable to remove service {}", e);
+        }
+    }
+
+    @Override
+    public void clean() {
+    }
+
+    @Override
+    public void start(final boolean wait) {
+        try {
+            ServiceCreateResponse serviceCreateResponse = dockerClient.createService(setServiceSpec());
+            if (wait) {
+                waitUntilServiceRunning().get(5, TimeUnit.MINUTES);
+            }
+            assertThat(serviceCreateResponse.id(), is(notNullValue()));
+        } catch (InterruptedException | DockerException | TimeoutException | ExecutionException e) {
```

Suppressing exceptions

---

throwing, done.